c/zss.c

```diff
@@ -988,13 +1032,113 @@ static bool readAgentHttpsSettings(ShortLivedHeap *slh,
   if (!address) {
     address = "127.0.0.1";
   }
-  bool httpsSettingsFound = port && settings->keyring;
-  if (httpsSettingsFound) {
+  bool isHttpsConfigured = port && settings->keyring;
+  if (settings->keyring) {
+      zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_INFO, ZSS_LOG_TLS_SETTINGS_MSG,
+              settings->keyring,
+              settings->label ? settings->label : "(no label)",
+              settings->password ? "****" : "(no password)",
+              settings->stash ? settings->stash : "(no stash)");
+    TlsEnvironment *tlsEnv = NULL;
+    int rc = tlsInit(&tlsEnv, settings);
+    if (rc != 0) {
+      zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_WARNING, ZSS_LOG_TLS_INIT_MSG, rc, tlsStrError(rc));
+    } else {
+      *outTlsEnv = tlsEnv;
+    }
+  }
+  if (isHttpsConfigured) {
     *outPort = port;
     *outAddress = address;
-    *outSettings = settings;
   }
-  return httpsSettingsFound;
+  return isHttpsConfigured;
+}
+
+#define NODE_MEDIATION_LAYER_SERVER_PREFIX   "ZWED_node_mediationLayer_server_"
+#define ENV_NODE_MEDIATION_LAYER_SERVER_KEY(key) NODE_MEDIATION_LAYER_SERVER_PREFIX key
+
+// Returns false if gateway host and port not found
+static bool readGatewaySettings(JsonObject *serverConfig,
+                                JsonObject *envConfig,
+                                char **outGatewayHost,
+                                int *outGatewayPort
+                               ) {
+  char *gatewayHost = jsonObjectGetString(envConfig, ENV_NODE_MEDIATION_LAYER_SERVER_KEY("hostname"));
```

Comment:
gatewayHostname also may exist to separate discovery host from gateway host.

Reply:
Fixed